Student/StudentE2ETests/Calendar/CalendarTests.swift

```diff
@@ -432,4 +432,119 @@ class CalendarTests: E2ETestCase {
         XCTAssertTrue(calendarEventItem.isVisible)
         XCTAssertTrue(titleLabel.isVisible)
     }
+
+    func testEditCalendarTodoItem() {
+        // MARK: Seed the usual stuff
+        let student = seeder.createUser()
+        let course = seeder.createCourse()
+        seeder.enrollStudent(student, in: course)
+        let title = "My dear calendar todo item"
+        let description = "Description of my dear calendar todo item."
+        let newTitle = "My edited todo item"
+
+        // MARK: Get the user logged in, navigate to Calendar
+        logInDSUser(student)
+        let calendarTab = Helper.TabBar.calendarTab.waitUntil(.visible)
+        XCTAssertTrue(calendarTab.isVisible)
+
+        calendarTab.hit()
+        let addButton = Helper.addNoteButton.waitUntil(.visible)
+        XCTAssertTrue(addButton.isVisible)
+
+        // MARK: Create Calendar Todo Item
+        addButton.hit()
+        let cancelButton = Helper.Todo.cancelButton.waitUntil(.visible)
+        let addButton2 = Helper.Todo.addButton.waitUntil(.visible)
+        let titleInput = Helper.Todo.titleInput.waitUntil(.visible)
+        let calendarSelector = Helper.Todo.calendarSelector.waitUntil(.visible)
+        let dateButton = Helper.Todo.dateButton.waitUntil(.visible)
+        let datePicker = Helper.Todo.datePicker.waitUntil(.visible)
+        var timePicker = Helper.Todo.timePicker.waitUntil(.visible)
+        let detailsInput = Helper.Todo.detailsInput.waitUntil(.visible)
+        XCTAssertTrue(cancelButton.isVisible)
+        XCTAssertTrue(addButton2.isVisible)
+        XCTAssertTrue(titleInput.isVisible)
+        XCTAssertTrue(calendarSelector.isVisible)
+        XCTAssertTrue(dateButton.isVisible)
+        XCTAssertTrue(datePicker.isVisible)
+        XCTAssertTrue(timePicker.isVisible)
+        XCTAssertTrue(detailsInput.isVisible)
+
+        titleInput.writeText(text: title)
+        calendarSelector.hit()
+        let newToDoButton = Helper.Todo.CalendarSelector.newToDoButton.waitUntil(.visible)
+        let userItem = Helper.Todo.CalendarSelector.userItem(user: student).waitUntil(.visible)
+        XCTAssertTrue(newToDoButton.isVisible)
+        XCTAssertTrue(userItem.isVisible)
+        XCTAssertTrue(userItem.hasValue(value: "0"))
+
+        userItem.hit()
+        XCTAssertTrue(userItem.waitUntil(.value(expected: "1")).hasValue(value: "1"))
+
+        newToDoButton.hit()
+        XCTAssertTrue(timePicker.waitUntil(.visible).isVisible)
+
+        timePicker.hit()
```

Comment:
Is this intentional, that `datePicker` is not tested for it's picker? (I saw it was the same for CreateToDo tests)

Reply:
Nice catch! Added with last commit. 👍